libs/sdk-ui-geo/src/core/geoChart/GeoChartLegendRenderer.tsx

```diff
@@ -154,25 +182,40 @@ function renderPushpinColorLegend(props: IGeoChartLegendRendererProps, hasColorL
         geoData,
         position = "top",
         responsive,
-        showFluidLegend,
+        isFluidLegend,
         colorLegendValue,
         numericSymbols = [],
+        contentRect,
     } = props;
 
     const data = geoData!.color!.data;
     const format = geoData!.color!.format;
+    const title = geoData?.size?.name;
 
     const dataWithoutNull = data.filter(isFinite);
     const colorData = generateLegendColorData(dataWithoutNull, colorLegendValue);
-    const isSmall: boolean = Boolean(responsive && showFluidLegend);
+    const width = contentRect?.client?.width;
+    const isAutoPosition = isAutoPositionWithPopup(responsive);
```

Comment:
what about to move all this micro logic to some helper functions easy to test, eg. 
getHeatmapLegendSize(...params): large|medium|small

Reply:
Done, check it please: https://github.com/gooddata/gooddata-ui-sdk/pull/1341/commits/8448bfe3f1903efa6639c4f50a462137021a94e9